docs/source/cpp/memory.rst

```diff
@@ -125,3 +125,38 @@ you can do so using the :class:`arrow::stl::allocator` wrapper.
 Conversely, you can also use a STL allocator to allocate Arrow memory,
 using the :class:`arrow::stl::STLMemoryPool` class.  However, this may be less
 performant, as STL allocators don't provide a resizing operation.
+
+Devices
+=======
```

Comment:
Would appreciate advice and suggestions on the wording here. cc @nealrichardson @wesm